src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -261,8 +261,16 @@ private static bool IsValidIssuer(string validIssuerTemplate, string tenantId, s
             }
         }
 
-        private BaseConfigurationManager GetEffectiveConfigurationManager(SecurityToken securityToken)
+        private BaseConfigurationManager GetEffectiveConfigurationManager(SecurityToken securityToken, TokenValidationParameters validationParameters)
         {
+            if (validationParameters.ConfigurationManager != null)
+            {
+                if (IsV2Authority)
+                    ConfigurationManagerV2 = validationParameters.ConfigurationManager;
+                else
+                    ConfigurationManagerV1 = validationParameters.ConfigurationManager;
+            }
+
             return (securityToken.Issuer.EndsWith(V2EndpointSuffix, StringComparison.OrdinalIgnoreCase)) ? ConfigurationManagerV2 : ConfigurationManagerV1;
         }
```

Comment:
This can be simplified to:

            if (validationParameters.ConfigurationManager != null)
                return validationParameters.ConfigurationManager;

            return (securityToken.Issuer.EndsWith(V2EndpointSuffix, StringComparison.OrdinalIgnoreCase)) ? ConfigurationManagerV2 : ConfigurationManagerV1;

The logic seems a bit confusing and the result could be incorrect.
Line 268 uses IsV2Authority to determine V1 or V2, but line 274 uses Issuer.EndsWith(V2EndpointSuffix).

If IsV2Authority  is true, then ConfigurationManagerV2 = validationParameters.ConfigurationManager.
If Issuer.EndsWith(V2EndpointSuffix) is false, then ConfigurationManagerV1 will be returned, instead of validationParameters.ConfigurationManager.

It seems we should use either IsV2Authority or Issuer.EndsWith(V2EndpointSuffix), not both. And Issuer.EndsWith(V2EndpointSuffix) is probably the one to use.